src/ngraph/ngraph_stats.cc

```diff
@@ -0,0 +1,144 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ngraph_stats.h"
+
+#include <iomanip>
+#include <string>
+#include "ngraph_utils.h"
+
+namespace ngraph_bridge {
+
+std::string exe_mode_to_string(int mode) {
+  switch (mode) {
+    case 0:
+      return "Inference";
+    case 1:
+      return "Train";
+    default:
+      return std::to_string(mode);
+  }
+}
+void NGraphStats::dump(std::ostream& out) {
+  if (ngraph_log_timer()) {
+    // accumulator for forward/backward/Combined summary at the end
+    const int pass_count = 3;
+    const std::string pass_name[pass_count] = {"Forward", "Backward",
+                                               "Combined"};
```

Comment:
I have changed where possible to use enums.

Reply:
I think you may have misunderstood my suggestion.  I was actually recommending defining a new enum that has exactly the members you need: forward, backward, and combined.

I hate to say it but IMO the change you made is actually a bit problematic, because it introduces an imperfect mapping between the elements of `pass_name` and the enumerators in `GraphExeMode`.